app/app.go

```diff
@@ -338,6 +338,18 @@ func NewAxelarApp(
 
 	app.SetAnteHandler(initAnteHandlers(encodingConfig, keys, keepers, appOpts))
 
+	// Register snapshot extensions to enable state-sync for wasm.
+	// MUST be done before loading the version
+	// Requires the snapshot store to be created and registered as a BaseAppOption
+	if manager := app.SnapshotManager(); manager != nil {
+		err := manager.RegisterExtensions(
+			wasmkeeper.NewWasmSnapshotter(app.CommitMultiStore(), getKeeper[wasm.Keeper](keepers)),
+		)
+		if err != nil {
+			panic(fmt.Errorf("failed to register snapshot extension: %s", err))
+		}
+	}
+
```

Comment:
I assume this should be kept behind a WASM_ENABLED flag check?

Reply:
Good catch